src/dispatch/static/dispatch/src/components/PreciseDateTimePicker.vue

```diff
@@ -0,0 +1,334 @@
+      </v-container>
+      <div class="mt-3 mb-3">Or paste a Unix timestamp:</div>
+      <v-text-field
+        class="paste-only"
+        v-model="unixTimestamp"
+        label="Unix Timestamp (UTC)"
+        readonly
+        @paste="handlePaste($event)"
+        @focus="highlightText"
+        @click="highlightText"
+      />
+    </v-card-text>
+    <v-card-actions>
+      <v-spacer />
+      <v-btn variant="text" @click="okHandler">Ok</v-btn>
+    </v-card-actions>
+  </v-card>
+</template>
+
+<script>
+import { parseISO } from "date-fns"
+import { formatInTimeZone } from "date-fns-tz"
+import moment from "moment-timezone"
+
+function isIsoDate(str) {
+  if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(str)) return false
+  const d = new Date(str)
+  return d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === str
+}
+
+function removeEndingZ(str) {
+  if (str.endsWith("Z")) {
+    str = str.slice(0, -1)
+  }
+  return str
+}
+
+export default {
+  props: {
+    modelValue: {
+      type: [Date, String],
+      default: null,
+    },
+    timezone: {
+      type: String,
+      default: null,
+    },
+  },
+
+  mounted() {
+    this.init()
+  },
+
+  data() {
+    return {
+      day: "00",
+      month: "00",
+      year: "0000",
+      hour: "00",
+      minutes: "00",
+      seconds: "00",
+      unixTimestamp: null,
+    }
+  },
+  computed: {
+    displayDateTime() {
+      if (this.date && this.time !== null && this.seconds !== null) {
+        return `${this.date} ${this.time}:${this.seconds}`
+      }
+      return ""
+    },
+    dateTimeWithDecimalSeconds() {
+      if (this.date && this.time !== null && this.seconds !== null) {
+        const [hours, minutes] = this.time.split(":")
+        return `${this.date}T${hours}:${minutes}:${this.seconds}`
+      }
+      return ""
+    },
+  },
+  methods: {
+    init() {
+      // given a timestamp in format yyyy-MM-dd'T'HH:mm:ss.SSS, parse into separate fields
+      if (this.modelValue) {
+        this.onTimestampChange(this.modelValue)
+        this.updateUnixTimestamp()
+      }
+    },
+    okHandler() {
+      let newValue = moment.tz(this.unixTimestamp, this.timezone).toISOString()
+      this.$emit("update:modelValue", newValue)
+      this.$emit("ok")
+    },
+    updateUnixTimestamp() {
+      // build Date object from fields
+      const date = `${this.year}-${this.month}-${this.day}T${this.hour}:${this.minutes}:${this.seconds}`
+      this.unixTimestamp = formatInTimeZone(date, "UTC", "yyyy-MM-dd'T'HH:mm:ss.SSS") + "Z"
+    },
+    handlePaste(event) {
+      event.preventDefault()
+      const clipboardData = event.clipboardData || window.clipboardData
+      let pastedData = clipboardData.getData("Text")
+      if (!pastedData.endsWith("Z")) {
+        pastedData += "Z"
+      }
+      // check to see if pastedData is a valid Unix timestamp
+      if (isIsoDate(pastedData)) {
```

Comment:
When I think of Unix Timestamp, I think of something like `1717457347.123` and it has no timezone - it is based on unix epoch (number of seconds since 01-Jan-1970). 

It seems like this is expecting one to use the ISO string format which would have a timezone as part of the string, but this only handles the UTC string of `Z` and not `+|-##:##`. If we were to support the ISO format, I am fine with that, but I would lean towards being expected to provide the timezone and have Dispatch parse it and interpret it correctly.